src/Dotnet.Script.DependencyModel/Environment/ScriptEnvironment.cs

```diff
@@ -30,11 +28,10 @@ private ScriptEnvironment()
             _installLocation = new Lazy<string>(GetInstallLocation);
             _platformIdentifier = new Lazy<string>(GetPlatformIdentifier);
             _runtimeIdentifier = new Lazy<string>(GetRuntimeIdentifier);
-            _isWindows = new Lazy<bool>(() => PlatformIdentifier == "win");
             _nuGetStoreFolder = new Lazy<string>(GetPathToNuGetStoreFolder);
         }
 
-        public bool IsWindows => _isWindows.Value;
+        public bool IsWindows => PlatformIdentifier == "win";
```

Comment:
Wouldn't checking the bool of an `Lazy<T>` be faster than doing string comparison every time the property is accessed?

Reply:
Yes, that's true and you can't beat the speed of a Boolean comparison. Best one can come close to is four Boolean comparisons:

```c#
public bool IsWindows
{
    get
    {
        var pid = PlatformIdentifier;
        return pid.Length == 3 && pid[0] == 'w' && pid[1] == 'i' && pid[2] == 'n';
    }
}
```

…which is slower by 40%. I didn't realize the purpose of making `IsWindows` lazy internally was micro-optimization. [`RuntimeEnvironment.OperatingSystemPlatform` is internally lazy](https://github.com/dotnet/core-setup/blob/9dff3ce12574699e6d4d31722047f775d92d05fd/src/managed/Microsoft.DotNet.PlatformAbstractions/Native/PlatformApis.cs#L18), then so is `ScriptEnvironment.PlatformIdentifier` and `ScriptEnvironment.IsWindows` on top of that, which seemed a bit excessive. Generally, if people are concerned about the performance of a property access/computation in a very tight loop then they'll cache the property access in a local variable on the stack.

Happy if you wish to keep things as-are and close this issue.